src/reducers/sources.js

```diff
@@ -138,7 +138,7 @@ function update(
     case "BLACKBOX":
       if (action.status === "done") {
         const url = action.source.url;
-        const isBlackBoxed = action.value.isBlackBoxed;
+        const { isBlackBoxed } = ((action: any): DonePromiseAction).value;
```

Comment:
This is a similar case to the other PR in which we can't tell from an `Action` if it has a `value` prop. I had hoped that my first take for `PromiseAction` would have allowed flow to see that when `status === 'done'` that `value` was valid but it wasn't able to (for reasons unknown). Now that I've simplified `PromiseAction` to avoid killing flow merges, there's no way to know since `value` is defined as `value?: type`.

So, we can either cast it to the `DonePromiseAction` or check for the existence of `value` before accessing it. While the former is a bit more verbose, it is (arguably) more clear and at least more explicit and avoids the existence check which does have a minimal run-time overhead.

I'd be okay with either approach but would like to be consistent in accessing `value` from a `PromiseAction`.

Reply:
lets follow up